### ISSUE TITLE
RAT: LightGBM requires at least two leaves and enough training rows and features

### ISSUE THREAD

**Issue by nm-z** [nm-z/recipe-dev#763]
## Error

```text
LightGBM requires at least two leaves and enough training rows and features
```

Reported while using `.lgbm().loss(&evaluator)` as the proposer in command RAT on `experimental/RAT`.

## Initialization mismatch

Command RAT constructs its proposal graph from one initial state/sample. Compiling an estimator block immediately calls its fitting routine. `fit_lightgbm` requires both a valid leaf count and at least `tree-min-rows` observations; the current default minimum is four.

The combined error does not identify which condition failed. In this initialization path, the one-row input is below that minimum, before the loop has collected measured observations. Increasing `.epochs()` cannot resolve an error raised before training starts.

CatBoost and XGBoost report their corresponding insufficient-row errors through the same path.

## Configuration

```rust
let evaluator = recipe.model().layer(16).tanh().loss(mse);
let proposal = recipe.model().lgbm().loss(&evaluator);
recipe.train()
    .rat(history, "./evaluate.lua")
    .epochs(100)
    .run(&proposal, &data);
```

The report arose in the locally modified stateful `recipe.data(auto)` flow. Those stateful changes are not yet committed; branch HEAD is `0cbc816c063d890447b27f231d3e96762bbb35b6`. The existing command-RAT graph construction also uses a one-row proposal initialization.

## Expected

RAT should support estimator initialization from real observations without trying to fit an ordinary supervised tree to a one-row placeholder. Do not hide the mismatch by duplicating rows, fabricating labels, or reducing the minimum-row requirement.

If a model composition is unsupported, report that specific limitation rather than implying the user's eventual observation corpus is too small.

## Source locations

In the inspected working tree: `fit_lightgbm` at `recipe.rs:13413`, `command_rat_graph` at `recipe.rs:7384`, and stateful proposal initialization near `recipe.rs:16501`. Line numbers include local changes. This issue records the user-reported error and source inspection, not a separately executed LightGBM reproduction.
